docs/topics/whatsnew/whatsnew-eap.md

```diff
@@ -25,6 +26,146 @@ All you need to do is to [change the Kotlin version](configure-build-for-eap.md)
+    // Uses `log` from the context
+    users.log("Log: $message")
+}
+
+// Declares a property with a context parameter
+context(users: UserService)
+val firstUser: String
+    // Uses `findUserById` from the context    
+    get() = users.findUserById(1)
+```
+
+You can use `_` as a context parameter name. In this case, the parameter's value is available for resolution but is not accessible by name inside the block:
+
+```kotlin
+// Uses `_` as context parameter name
+context(_: UserService)
+fun logWelcome() {
+    // Resolution still finds the appropriate `log` function from UserService
+    outputMessage("Welcome!")
+}
+```
+
+#### Context parameters resolution
+
+Kotlin resolves context parameters at the call site by searching for matching context values in the current scope. Kotlin matches them by their type.
+If multiple compatible values exist at the same scope level, the compiler reports an ambiguity:
+
+```kotlin
+// `UserService` defines the dependency required in context
+interface UserService {
+    fun log(message: String)
+}
+
+// Declares a function with a context parameter
+context(users: UserService)
+fun outputMessage(message: String) {
+    users.log("Log: $message")
+}
+
+fun main() {
+    // Implements `UserService` 
+    val serviceA = object : UserService {
+        override fun log(message: String) = println("A: $message")
+  }
+
+  // Implements `UserService`
+  val serviceB = object : UserService {
+      override fun log(message: String) = println("B: $message")
+  }
+
+  // Both `serviceA` and `serviceB` match the expected `UserService` type at the call site
+  context(serviceA, serviceB) {
+      outputMessage("This will not compile")
+      // Ambiguity error
+  }
+}
+```
+
+#### Restrictions
+
+Context parameters are in continuous improvement; some of the current restrictions are:
+
+* Constructors cannot declare context parameters
+* Properties with context parameters can't have backing fields or initializers
+* Properties with context parameters can't use delegation
+
+However, context parameters in Kotlin represent a significant improvement in managing dependencies through simplified dependency injection,
+improved DSL design, and scoped operations. For more information, see the feature's [KEEP](https://github.com/Kotlin/KEEP/blob/context-parameters/proposals/context-parameters.md).
+
+#### How to enable context parameters
+
+To enable context parameters in your project, use the following compiler option in the command line:
+
+```Bash
+-Xcontext-parameters
+```
+
+Or add it to the `compilerOptions {}` block of your Gradle build file:
+
+```kotlin
+// build.gradle.kts
+kotlin {
+    compilerOptions {
+        freeCompilerArgs.add("-Xcontext-parameters")
+    }
+}
+```
+
+> Specifying both `-Xcontext-receivers` and `-Xcontext-parameters` compiler options simultaneously leads to an error.
+>
+{style="warning"}
+
+#### Leave your feedback
+
+We plan to stabilize and improve this feature in future Kotlin releases.
```

Comment:
Just a thought: there are too many "we" in the text about the Language updates.

Reply:
Great catch. I replaced some.